server/clip_server/model/tokenization.py

```diff
@@ -23,13 +30,19 @@ def __call__(
     ):
         """
         :param texts: An input string or a list of input strings to tokenize
-        :param context_length: The context length to use; all CLIP models use 77 as the context length.
+        :param context_length: The context length to use; all English CLIP models use 77 as the context length.
+            for Chinese CLIP models, context_length = 52, no truncate
```

Comment:
will this cause memory issue if no truncate is used?

Reply:
I had similar concern and searched from their repo, [here](https://github.com/OFA-Sys/Chinese-CLIP/commit/cfe232dcd436b95d5ea728b2eee682f9146efc84) they adjusted the context_length from 24 to 52 to avoid truncation when training, maybe based on assume that it's hard to find a 50-Chinese-letter sentence. And in tokenize function there is no more truncate [here](https://github.com/OFA-Sys/Chinese-CLIP/blob/bfd8d0665bdee1e9b3d5908f9392c4c8b2f8b5bc/cn_clip/clip/utils.py#L120)

Reply:
It seems that they truncate the sentence to context_length, as you can see in their tokenizer: [here](https://github.com/OFA-Sys/Chinese-CLIP/blob/bfd8d0665bdee1e9b3d5908f9392c4c8b2f8b5bc/cn_clip/clip/utils.py#L138) and [here](https://github.com/OFA-Sys/Chinese-CLIP/blob/bfd8d0665bdee1e9b3d5908f9392c4c8b2f8b5bc/cn_clip/clip/utils.py#L141)

Reply:
I'm OK with this pr if there is no memory issues here

Reply:
Actually I tried with their code, this is the result for the code  `text = clip.tokenize(["妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子妙蛙种子"]).to(device)`   (64 characters in this sentence):

`tensor([[ 101, 1975, 6032, 4905, 2094, 1975, 6032, 4905, 2094, 1975, 6032, 4905,
         2094, 1975, 6032, 4905, 2094, 1975, 6032, 4905, 2094, 1975, 6032, 4905,
         2094, 1975, 6032, 4905, 2094, 1975, 6032, 4905, 2094, 1975, 6032, 4905,
         2094, 1975, 6032, 4905, 2094, 1975, 6032, 4905, 2094, 1975, 6032, 4905,
         2094, 1975, 6032,  102]], device='cuda:0')`

The result is same when using CaS, seems they directly truncate and omit characters left.